src/edit_view.rs

```diff
@@ -59,20 +82,142 @@ const TOP_PAD: f32 = 6.0;
+        for line_num in first_line..last_line {
+            if let Some(textline) = self.get_text_line(line_num) {
+                textline.draw_text(rt, x0, y, &resources.fg);
+                textline.draw_cursor(rt, x0, y, &resources.fg);
+            }
+            y += LINE_SPACE;
+        }
+        self.resources = Some(resources);
+    }
+
+    fn layout(&mut self, bc: &BoxConstraints, _children: &[Id], _size: Option<(f32, f32)>,
+        _ctx: &mut LayoutCtx) -> LayoutResult
+    {
+        LayoutResult::Size(bc.constrain((0.0, 0.0)))
+    }
+
+    fn mouse(&mut self, event: &MouseEvent, _ctx: &mut HandlerCtx) -> bool { 
+        let MouseEvent { x, y, mods: _, which, count } = *event;
+        if which == MouseButton::Left && count == 1 {
+            let (line, col) = self.xy_to_line_col(x, y);
+            let params = json!({
+                "ty": "point_select",
+                "line": line,
+                "col": col,
+            });
+            self.send_edit_cmd("gesture", &params);
+        }
+        false
+    }
+
+    fn mouse_moved(&mut self, _x: f32, _y: f32, _ctx: &mut HandlerCtx) {}
+
+    fn on_hot_changed(&mut self, _hot: bool, _ctx: &mut HandlerCtx) {}
+
+    fn poke(&mut self, payload: &mut Any, ctx: &mut HandlerCtx) -> bool { 
+        if let Some(cmd) = payload.downcast_ref::<EditViewCommands>() {
+            match cmd {
+                EditViewCommands::ViewId(view_id) => {
+                    self.view_id = view_id.to_string();
+                }
+                EditViewCommands::ApplyUpdate(update) => {
+                    self.apply_update(&update);
+                }
+                EditViewCommands::ScrollTo(line) => {
+                    self.scroll_to(*line);
+                }
+                EditViewCommands::Core(core) => {
+                    self.core = core.clone();
+                }
+                EditViewCommands::Undo => {
+                    self.send_action("undo");
+                }
+                EditViewCommands::Redo => {
+                    self.send_action("redo");
+                }
+                EditViewCommands::UpperCase => {
+                    self.send_action("uppercase");
+                }
+                EditViewCommands::LowerCase => {
+                    self.send_action("lowercase");
+                }
+                EditViewCommands::Transpose => {
+                    self.send_action("transpose");
+                }
+                EditViewCommands::AddCursorAbove => {
+                    self.add_cursor_above();
+                }
+                EditViewCommands::AddCursorBelow => {
+                    self.add_cursor_below();
+                }
+                EditViewCommands::SingleSelection => {
+                    self.single_selection();
+                }
+                EditViewCommands::SelectAll => {
+                    self.select_all();
+                }
+            }
+            ctx.invalidate();
```

Comment:
This is probably too aggressive, but it doesn't make much difference now. Commands that cause edits don't need to invalidate, because the update from the core will cause invalidation. Of course, `apply_update` and `scroll_to` do need to invalidate.

At some point, we'll get fine-grained invalidation, which will be a good opportunity to revisit this.

So I'm basically ok with this as is, but would add "TODO: finer grained invalidation".